src/assets/ts/components/line.tsx

```diff
@@ -21,6 +22,7 @@ export type LineProps = {
   wordHook?: PartialUnfolder<Token, React.ReactNode>;
   onCharClick?: ((col: Col, e: Event) => void) | undefined;
   cursorBetween?: boolean;
+  session?: Session;
```

Comment:
might be better to not have the Session abstraction here and just pass in something like 

renderLineAnnotationHook?: (info: { lineData: .., column: ..., cursors: ...}) => Array<React.ReactNode>,

Reply:
I got it working that way, but I had to bind the session to the function. Does that defeat the purpose? Genuinely curious exactly why this way is cleaner.

Reply:
yeah, fine to bind the session to the function from the caller.  i don't know if it really matters, was just going by the principle of require less abstractions in interfaces :)

Reply:
makes sense, thanks for explaining!